start/software/compiling-applications.md

```diff
@@ -1,57 +1,135 @@
+useful information for compiling and using your software in OSG Connect. A detailed 
+example of performing software compilation is additionally available 
+at (link). 
+
+
+> *What is compiling?*
+> The process of compiling converts human readable code into binary, 
+> machine readable code that will execute the steps of the program. 
+
+## Get software source code
+
+The first step to compiling your software is to locate and download the source 
+code, being sure to select the version that you want. Source code 
+will often be made available as a compressed tar archive which will need to be 
+extracted for before compilation.
+
+You should also carefully review the installation instructions provided by the 
+software developers. The installation instructions should include important 
+information regarding various options for configuring and performing the compilation. 
+Also carefully note any system dependencies (hardware, other software, and libraries) that are 
+required for your software.
+
+## Select the appropriate compiler and compilation options
+
+A compiler is a program that is used to peform source code compilation. The GNU Compiler 
+Collection (GCC) is a common, open source collection of compilers with support for C, C++, 
+fotran, and other languages, and includes important libraries for supporting your compilation 
+and sometimes software executation. Your software compilation may require certain versions 
+of a compiler which should be noted in the installtion instructions or system dependencies 
+documention. Currently the login nodes have `GCC 4.8.5` as the default version, but newer 
+versions of GCC may also be available - to learn more please contact <support@osgconnect.net>.
+
+CMake is a commonly used compilation platform. Your software may have dependencies for 
+specific `cmake` versions. Currently the login nodes have two versions of CMake, `3.12.3` 
+and `3.13.0` available as [modules](https://support.opensciencegrid.org/support/solutions/articles/12000048518). 
+
+### Static versus dynamic linking during compilation
+
+Binary code often depends on additional information (i.e. instructions) from other software, 
+known as libraries, for proper execution. The default behavior when compiling, is for the 
+final binary to be "dynamically linked" to libraries that it depends on, such that when 
+the binary is executed, it will look for these library files on the system that it is 
+running on. Thus a copy of the appropriate library files will need to be available to your 
+software wherever it runs. OSG Connect users can transfer a copy of the necessary 
+libraries alog with with their jobs to manage such dependencies if not supported by the 
+execute node that your jobs run on.
+
+However, the option exists to "statically link" the library dependencies of your software. 
+By statically linking libraries during compilation, the library code will be 
```

Comment:
Combine the two sentences starting on line 57.

Reply:
i have revise to reduce redundancy